src/RowContainer.js

```diff
@@ -0,0 +1,44 @@
+import React, {PropTypes} from 'react';
+import {ContextMenuLayer} from 'react-contextmenu';
+
+class RowContainer extends React.Component {
+  hasContextMenu() {
+    return this.props.contextMenu && React.isValidElement(this.props.contextMenu);
+  }
+
+  renderRowsWithContextMenu() {
+    let newProps = {rowIdx: this.props.rowIdx, idx: this.props.idx};
+    let contextMenu = React.cloneElement(this.props.contextMenu, newProps);
```

Comment:
not sure what recommended practice is for react-context-menu, but here we will be doing the clone on every row-render, which seems expensive?